embedded/document/engine.go

```diff
@@ -978,3 +978,8 @@ func (d *documentReader) Read(ctx context.Context, count int) ([]*structpb.Struc
 func (d *documentReader) Close() error {
 	return d.reader.Close()
 }
+
+// CopyCatalogToTx copies the current sql catalog to the ongoing transaction.
+func (e *Engine) CopyCatalogToTx(ctx context.Context, tx *store.OngoingTx) error {
+	return e.sqlEngine.CopyCatalogToTx(ctx, tx)
```

Comment:
isn't it covered with the sql catalog already?

note: is the sql catalog copied with the indexing up to date?

Reply:
document sql engine is instantiated with a different prefix